utils/user_info.py

```diff
@@ -191,3 +193,27 @@ def send_forget_password_email(request, user):
         Thread(target=msg.send, args=()).start()
     except:
         logger.warning("There is an error when sending email to %s's mailbox" % username)
+
+def get_public_users():
+    return User.objects.filter(username__startswith = ANONYMOUS_PREFIX)
+
+def attends_not_ended_contest(user):
+    user_attends = Contestant.objects.filter(user = user)
+    for contestant in user_attends:
```

Comment:
sorry, I had some misunderstanding.
Then I suggest that you can try [django select_related filter](https://docs.djangoproject.com/en/1.7/ref/models/querysets/#select-related)